src/runtime/test/json-id-to-manifest-test.ts

```diff
@@ -0,0 +1,120 @@
+/** @see LICENSE */
+
+import {assert} from '../../platform/chai-web.js';
+import {JsonldToManifest, supportedTypes} from '../converters/jsonldToManifest.js';
+import {fetch} from '../../platform/fetch-web.js';
+import {Predicate} from '../hot.js';
+import {Manifest} from '../manifest.js';
+
+describe('JsonldToManifest', () => {
+
+  const isValidManifest: Predicate<string> = (manifestStr: string): boolean => {
+    try {
+      Manifest.parse(manifestStr);
+    } catch (error) {
+      return false;
+    }
+    return true;
+  };
+
+  const asyncProductStr: Promise<string> = fetch('https://schema.org/Product.jsonld')
+    .then(r => r.text());
+
+  const asyncLocalBusinessStr: Promise<string> = fetch('https://schema.org/LocalBusiness.jsonld')
```

Comment:
Thank you, this is great feedback. 

I've made a few changes based on this: 
I've decided to download the jsonld files into an `assets` folder from tests. That way, we won't have any external dependencies. This is also cleaner than having a giant string in the middle of a test. I'm choosing real jsonld files since I think it exercises more code than contrived examples I can come up with. 

I've also converted the constants into a single function that will only load assets when needed. I'd like to use functions instead of `before` since I want to use different data for different tests. Further, I think it makes things easier to parameterize if things are inside functions instead of in a stateful `before`. This is most likely a style choice, and if you instant, I'm happy to change this pattern to use those that are conventional to the code base.

Reply:
Ok, thanks for updating. It's perfectly fine to use "custom" functions instead of the `before` hook, and definitely better to not have a stateful `before`.